spec/classes/init_spec.rb

```diff
@@ -191,12 +190,219 @@
             'enable'     => true,
             'hasrestart' => true,
             'hasstatus'  => v[:service_hasstatus],
+            'subscribe'  => [
+              'File[/var/lib/monit]',
+              'File[monit_config_dir]',
+              'File[monit_config]',
+            ]
           })
         end
       end
     end
   end
 
+  describe 'parameter functionality' do
+    let(:facts) do
+      {
+       :osfamily                  => 'Debian',
+       :lsbdistcodename           => 'squeeze',
+      }
+    end
+
+    context 'when check_interval is set to valid integer <242>' do
+      let(:params) { { :check_interval => 242 } }
+      it { should contain_file('monit_config').with_content(/^set daemon 242$/) }
+    end
+
+    context 'when httpd is set to valid bool <true>' do
+      let(:params) { { :httpd => true } }
+      content = <<-END.gsub(/^\s+\|/, '')
```

Comment:
I don't like that using %{} you loose proper indenting and therefore disturbs readability. Which is kept with the help of gsub().

For fixtures I have the same feeling like for params.pp ;)
I like them for a fully feature test to ensure all formating and sequencing details. Afterwards I prefer to only test the changed content, as long as we are talking about less than lets say 5 or 6 lines.